lxc/network.go

```diff
@@ -146,14 +146,26 @@ func (c *cmdNetworkAttach) Run(cmd *cobra.Command, args []string) error {
 	}
 
 	// Prepare the instance's device entry
-	device := map[string]string{
-		"type":    "nic",
-		"nictype": "macvlan",
-		"parent":  resource.name,
-	}
+	var device map[string]string
+	if network.Managed {
```

Comment:
I think you also need an HasExtension check there as older LXD releases will have managed bridges but no network= property.

Reply:
Good point, this has been updated.